apps/src/templates/AppView.jsx

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var ProtectedStatefulDiv = require('./ProtectedStatefulDiv.jsx');
+
+/**
+ * Top-level React wrapper for our standard blockly apps.
+ */
+var AppView = React.createClass({
+  propTypes: {
+    renderCodeApp: React.PropTypes.func.isRequired,
+    onMount: React.PropTypes.func.isRequired
+  },
+
+  componentDidMount: function () {
+    this.refs.codeApp.getDOMNode().innerHTML = this.props.renderCodeApp();
+    this.props.onMount();
+  },
+
+  render: function () {
+    return <ProtectedStatefulDiv ref="codeApp" />;
+  }
```

Comment:
I'm not saying the validation is a bad idea, I just can't think of a way to make it automatic enough to catch all cases.  Here's some brainstorming:

* The top-level view component could wrap all of its ejs rendering functions and verify that they've been called exactly once in `componentDidMount`.  That still requires the dev to wrap any new ejs rendering function they add to the top-level propTypes, but that seems less likely to get missed.

* Instead of wrapping those functions, this concept might be easy to implement as a unit test, passing in fake render functions that sense whether they've been called, then rendering the view with as many property permutations as possible.  But again, vulnerable to developer error.

* I wonder if there's some magic way to make a component report 'hasStatefulChildren' if it requires ProtectedStatefulDiv.jsx or any other component that reports `hasStatefulChildren` is true.  Then you could do some validation that any component that requires a stateful child renders at least one.  That doesn't help if a component has multiple stateful children though.